### DISCUSSION TITLE
什么时候会出现Conflict session导致服务器和客户端断开链接

### DISCUSSION THREAD

**Discussion by ericzhanchina** [cloudwu/skynet#1782] [General]
在日志里面发现了部分Conflict session的错误，看了一下MsgServer的代码，不是特别明白为何出现了Conflict session的错误。客户端向服务器发送session一样的请求，好像并不会出现Conflict session。请指点。

**Comment by cloudwu** [comment]
同一个连接不应该发送一样的 session ，因为连接并没有断开。即，不应该在连接有效时，重复同一个请求。因为此时连接没断开，如果请求没有回复只表示服务器还在处理中，不要着急再次请求。

重复请求只应发生在不同的连接上，即上一个连接断开了，所以不确定某个 session 上的 response 是不是发丢了。

**Reply by ericzhanchina** [reply]
但是我测试了一下，设置了一个心跳协议，故意不增加session的数值，不断发一样session的值，并没有抛出Conflict session的错误

**Reply by cloudwu** [reply]
读代码，server 并不会永久保存所有历史上的 session 。检查 session 是否和历史上有过的任意一个重复是不可能的。

**Reply by ericzhanchina** [reply]
谢谢云风，我仔细看看
